### PR TITLE
Add 'fleet add-device' and 'device extract'.

### PR DESCRIPTION
**Pull request by floitsch**
Continue to rename "create" to "add".

### CODE REVIEW

File: tests/cmd-fleet-add-devices-test.toit [hunk 1]
```diff
@@ -45,40 +45,41 @@ run-test test-cli/TestCli fleet-dir/string:
...
         --name="test-name"
         --aliases=aliases
 
     test-cli.run --expect-exit-1 --allow-exception [
       "fleet",
-      "create-identity",
-      "--output-directory", tmp-dir,
-      "$id",
+      "add-device",
+      "--output", "$tmp-dir/other.identity",
```
Comment (kasperl) [inline tests/cmd-fleet-add-devices-test.toit:64]: No need for format? Or is this supposed to fail with exit code 1 here. Probably.

File: tests/base-qemu/base.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-version: 1
+$schema: https://toit.io/schemas/artemis/pod-specification/v1.json
```
Comment (kasperl) [inline tests/base-qemu/base.yaml:1]: Did we already update this? I don't think so. We should push the new one out (the one that doesn't have chip support).

File: src/cli/cmds/utils_.toit [hunk 3]
```diff
@@ -68,3 +68,9 @@ compute-fleet-root-or-ref parsed/cli.Parsed config/Config ui/Ui -> string:
       printer.emit "Using fleet-root '$fleet-env' provided by environment variable."
     return fleet-env
   return "."
+
+make-default_ --device-id/uuid.Uuid --config/Config --ui/Ui:
+  config[CONFIG-DEVICE-DEFAULT-KEY] = "$device-id"
+  config.write
+  ui.info "Default device set to $device-id."
+
```
Comment (kasperl) [inline src/cli/cmds/utils_.toit:76]: Drop last newline.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/cli/cmds/device.toit [hunk 4]
```diff
@@ -375,36 +355,96 @@ set-max-offline parsed/cli.Parsed config/Config cache/Cache ui/Ui:
...
     ui.info "Firmware successfully written to '$output'."
 
-build-image device/Device pod/Pod --tar/bool --output/string --cache/Cache --ui/Ui:
-    firmware := Firmware --cache=cache --device=device --pod=pod
-    sdk := get-sdk pod.sdk-version --cache=cache
-    with-tmp-directory: | tmp-dir/string |
-      device-specific-path := "$tmp-dir/device-specific"
-      device-specific := firmware.device-specific-data
-      file.write-content --path=device-specific-path device-specific
+check-pod-is-esp32_ --pod/Pod --sdk/Sdk --ui/Ui:
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/cli/cmds/device.toit]: I had forgotten to call this function...
It's only used once, so I inlined it.
done.

File: src/cli/cmds/fleet.toit [hunk 5]
```diff
@@ -105,100 +107,56 @@ create-fleet-commands config/Config cache/Cache ui/Ui -> List:
...
+      --aliases=["create-device"]
       --help="""
-        Create a single identity file.
+        Add a new device to the fleet.
 
-        An identity file describe a device, containing their ID and organization.
-        For each written identity file, a device is provisioned in the Toit
-        cloud.
+        If no output file is given, the device is just added to the fleet. In that
+        case you can use the 'device' command to extract an image, or you can
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/cli/cmds/fleet.toit]: done.

File: src/cli/cmds/fleet.toit [hunk 6]
```diff
@@ -105,100 +107,56 @@ create-fleet-commands config/Config cache/Cache ui/Ui -> List:
...
+        use 'flash' to flash it.
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/cli/cmds/fleet.toit]: done.

File: src/cli/cmds/fleet.toit [hunk 7]
```diff
@@ -105,100 +107,56 @@ create-fleet-commands config/Config cache/Cache ui/Ui -> List:
...
+            --help="The id of the device.",
+        cli.Flag "default"
+            --default=true
+            --help="Make this device the default device.",
       ]
       --examples=[
+        cli.Example "Add a new device in group 'roof-solar':"
+            --arguments="--group=roof-solar",
+        cli.Example "Add a new device and write the identity file to 'device.identity':"
+            --arguments="-o device.identity",
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/cli/cmds/fleet.toit]: good catch.
done.

File: src/cli/cmds/utils_.toit (hunk 3 above)
Comment (floitsch) [reply src/cli/cmds/utils_.toit]: done.

File: tests/base-qemu/base.yaml (hunk 2 above)
Comment (floitsch) [reply tests/base-qemu/base.yaml:1]: it still has the "chip". We need to update it.

File: tests/cmd-fleet-add-devices-test.toit (hunk 1 above)
Comment (floitsch) [reply tests/cmd-fleet-add-devices-test.toit:66]: The 'format' was missing.
I added a comment what we are testing.
done.

File: tests/utils.toit [hunk 8]
```diff
@@ -655,7 +655,24 @@ class TestDevicePipe extends TestDevice:
...
+      // Running on Windows is tricky...
+      // - We want Git's bash and not any other. The path to the git-bash must
+      //   be in Windows format.
+      // - We provide a shell script as argument. That script must be in Unix
+      //   (cygwin) format.
+      // Example: `C:/Program Files/Git/usr/bin/bash.exe /tmp/envelope-test-...`.
+      script-path = (pipe.backticks ["cygpath", "-u", script-path]).trim
+      program-files-path := os.env.get "ProgramFiles"
+      if not program-files-path:
+        // This is brittle, as Windows localizes the name of the folder.
```
Comment (kasperl) [inline tests/utils.toit:672]: Maybe extend the comment to explain why it is good enough for our testing?

Comment (floitsch) [reply tests/utils.toit]: Removed it again.
We can't nicely kill the subprocesses on Windows, so I just skip them.

**Pull request merged**